proxy/deploy.go

```diff
@@ -21,10 +21,26 @@ type FunctionResourceRequest struct {
 	Requests *stack.FunctionResources
 }
 
-// DeployFunction call FaaS server to deploy a new function
 func DeployFunction(fprocess string, gateway string, functionName string, image string,
 	language string, replace bool, envVars map[string]string, network string,
-	constraints []string, update bool, secrets []string, labels map[string]string, functionResourceRequest1 FunctionResourceRequest) {
+	constraints []string, update bool, secrets []string, labels map[string]string,
+	functionResourceRequest1 FunctionResourceRequest) {
+
+	statusCode := DeployFunctionImpl(fprocess, gateway, functionName, image, language, replace, envVars, network, constraints, update, secrets, labels, functionResourceRequest1)
```

Comment:
Not keen on names like "Impl".. we can just call this `Deploy`? I saw we broke out a function for this - could we make the change one level higher? Does that even make sense?

Reply:
By make the change one level higher -- do you mean having the retry logic (retry if 404 error occurs) in `commands/deploy.go`, in the `runDeploy()` function?